ldpd/ldpd.h

```diff
@@ -531,13 +532,18 @@ struct kroute {
 };
 
 struct kpw {
+	char                     ifname[IF_NAMESIZE];
 	unsigned short		 ifindex;
 	int			 pw_type;
+	struct in_addr           lsr_id;
 	int			 af;
 	union ldpd_addr		 nexthop;
 	uint32_t		 local_label;
 	uint32_t		 remote_label;
 	uint8_t			 flags;
+	uint32_t                 pwid;
+	char			 vpn_name[L2VPN_NAME_LEN];
+	unsigned short           ac_port_ifindex;
 };
```

Comment:
What is the `ac_port_ifindex` for? It's not being used anywhere.

Also, the other members you're adding here are not necessary in my opinion.

We don't want to send protocol-specific pseudowire attributes to zebra, like `pwid`, `lsr_id` and `vpn_name`. Currently only ldpd supports L2VPNs, but in the future we should implement support for BGP-signaled L2VPNs as well.

I think that pseudowires can be uniquely identified by their ifname and ifindex, zebra doesn't need to track more information than that.